File: types/module.ts
/**
 * 模块系统类型定义
 */

export interface ModuleDefinition {
  id: number
  moduleKey: string
  moduleName: string
  moduleVersion: string
  description?: string
  author?: string
  icon?: string
  category?: string
  dependencies?: string[]
  routes?: string[]
  components?: string[]
  permissions?: Record<string, any>
  configSchema?: Record<string, any>
  isEnabled: boolean
  isCore: boolean
  sort: number
  createdAt: string
  updatedAt: string
}

export interface ModuleConfig {
  id: number
  moduleKey: string
  configKey: string
  configValue: any
  description?: string
  createdAt: string
  updatedAt: string
}

export interface ModuleRoute {
  path: string
  name?: string
  component?: string
  meta?: Record<string, any>
  children?: ModuleRoute[]
}

export interface ModuleComponent {
  name: string
  path: string
  global?: boolean
  lazy?: boolean
}

export interface ModuleManifest {
  moduleKey: string
  moduleName: string
  version: string
  description?: string
  author?: string
  icon?: string
  category?: string
  dependencies?: string[]
  routes?: ModuleRoute[]
  components?: ModuleComponent[]
  permissions?: Record<string, any>
  configSchema?: Record<string, any>
  entry?: string
}

export interface ModulePermission {
  key: string
  name: string
  description: string
  level: 'basic' | 'advanced' | 'admin'
}

export interface ConfigSchemaProperty {
  type: 'string' | 'number' | 'boolean' | 'object' | 'array'
  required?: boolean
  default?: any
  description?: string
  min?: number
  max?: number
  pattern?: string
  enum?: string[]
}

export interface CreateModuleRequest {
  moduleKey: string
  moduleName: string
  description?: string
  author?: string
  icon?: string
  category: string
  dependencies?: string[]
  routes?: ModuleRoute[]
  components?: ModuleComponent[]
  permissions?: ModulePermission[]
  configSchema?: Record<string, ConfigSchemaProperty>
  sort?: number
}

export interface UpdateModuleRequest extends Partial<CreateModuleRequest> {
  version?: string
  isEnabled?: boolean
}

export interface ModuleDependencyGraph {
  [key: string]: {
    dependsOn: string[]
    dependents: string[]
  }
}

export interface ModuleStats {
  totalModules: number
  enabledModules: number
  disabledModules: number
  coreModules: number
  categoryDistribution: Record<string, number>
  recentDownloads: number
  avgRating: number
}

export interface ModuleVersion {
  id: number
  moduleKey: string
  version: string
  packageUrl: string
  packageSize: number
  checksum: string
  changelog?: string
  isLatest: boolean
  isStable: boolean
  publishedAt: string
  createdBy?: string
  createdAt: string
  updatedAt: string
}

export interface ModuleDownload {
  id: number
  moduleKey: string
  version: string
  userId?: number
  ipAddress: string
  userAgent?: string
  downloadedAt: string
}

export interface ModuleReview {
  id: number
  moduleKey: string
  version: string
  userId: number
  rating: number
  title?: string
  content?: string
  isVerified: boolean
  createdAt: string
  updatedAt: string
}

export interface ModuleCompatibility {
  id: number
  moduleKey: string
  version: string
  nuxtVersionMin?: string
  nuxtVersionMax?: string
  nodeVersionMin?: string
  browserCompatibility?: {
    chrome?: string
    firefox?: string
    safari?: string
    edge?: string
  }
  dependencies?: {
    [key: string]: string
  }
  createdAt: string
  updatedAt: string
}

export interface UploadModuleRequest {
  moduleKey: string
  version: string
  file: File
  changelog?: string
  isStable?: boolean
}

export interface CreateVersionRequest {
  moduleKey: string
  version: string
  changelog?: string
  isStable?: boolean
}

export interface VersionDiff {
  added: string[]
  removed: string[]
  changed: string[]
  renamed: string[]
}

export interface ModuleRating {
  averageRating: number
  totalReviews: number
  distribution: {
    5: number
    4: number
    3: number
    2: number
    1: number
  }
}

File: backend/PersonalSite.Api/Models/Module.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PersonalSite.Api.Models;

[Table("module")]
public class Module
{
    [Key]
    [Column("id")]
    public long Id { get; set; }

    [Required]
    [MaxLength(50)]
    [Column("module_key")]
    public string ModuleKey { get; set; } = string.Empty;

    [Required]
    [MaxLength(100)]
    [Column("module_name")]
    public string ModuleName { get; set; } = string.Empty;

    [MaxLength(20)]
    [Column("module_version")]
    public string ModuleVersion { get; set; } = "1.0.0";

    [Column("description", TypeName = "TEXT")]
    public string? Description { get; set; }

    [MaxLength(100)]
    [Column("author")]
    public string? Author { get; set; }

    [MaxLength(100)]
    [Column("icon")]
    public string? Icon { get; set; }

    [MaxLength(50)]
    [Column("category")]
    public string? Category { get; set; }

    [Column("dependencies", TypeName = "TEXT")]
    public string? Dependencies { get; set; }

    [Column("routes", TypeName = "TEXT")]
    public string? Routes { get; set; }

    [Column("components", TypeName = "TEXT")]
    public string? Components { get; set; }

    [Column("permissions", TypeName = "TEXT")]
    public string? Permissions { get; set; }

    [Column("config_schema", TypeName = "TEXT")]
    public string? ConfigSchema { get; set; }

    [Column("is_enabled")]
    public bool IsEnabled { get; set; } = true;

    [Column("is_core")]
    public bool IsCore { get; set; } = false;

    [Column("sort")]
    public int Sort { get; set; }

    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.Now;

    [Column("updated_at")]
    public DateTime UpdatedAt { get; set; } = DateTime.Now;

    // 导航属性
    public virtual ICollection<ModuleConfig> ModuleConfigs { get; set; } = new List<ModuleConfig>();
}

